dotmap/test.py

```diff
@@ -383,3 +383,21 @@ def test_subclass_with_property(self):
         p.my_prop.second.third = 456
         self.assertIsInstance(p.my_prop.second, PropertyDotMap)
         self.assertEqual(p.my_prop.second.third, 456)
+
+class TestKeyConvertHook(unittest.TestCase):
+    def fix_illegal_key(self, key):
+        return key.replace(".", "_").replace("-","_")
```

Comment:
Okay. It may be the latest master but I don't see how this can function as you are expecting. You only do the replacement in the tests, not the main code in `__init__.py`. So while the test passes, the main code isn't functional.

Reply:
Actually, I did some changes in main code __init__.py, this pull request includes two files change, please checkout.

Reply:
I see. So you've changed this from a simple flag to a custom function. Good call.

Reply:
create another pull request #90 to support applying this hook on nested fields, please have a check.